midp/gfx.js

```diff
@@ -405,8 +407,8 @@ var currentlyFocusedTextEditor;
     };
 
     Native["javax/microedition/lcdui/Font.charWidth.(C)I"] = function(char) {
-        withFont(this, MIDP.context2D);
-        return measureWidth(MIDP.context2D, String.fromCharCode(char));
+        tempContext.font = this.css;
```

Comment:
withFont was optimized by avoiding to set the *font* property if not necessary. Setting it was somehow slower (IIRC @yxl implemented this optimization)